tests/test_result.py

```diff
@@ -0,0 +1,182 @@
+from unittest.mock import MagicMock
+import pathlib
+import django
+from tcms_django_plugin import TestRunner
+from testapp.tests import (TestAppAboutPageTest, TestAppMissingPageTest,
+                           TestAppContactPageTest, TestAppIndexPageTest,
+                           TestAppBooleanPageTest)
+
+
+def setup(test_case, method_name):
```

Comment:
Use Django's test format and start the tests via `manage.py test`, see
https://docs.djangoproject.com/en/3.0/topics/testing/overview/ and https://github.com/kiwitcms/tenants for example.

Reply:
I had initially tried something using Django's testing instruments via `manage.py test` but all the solutions I could come up to test this runner and the Testresult class with ended up adding up unnecessary complexity when trying to mock up the API and ended up not using any of the features Django's testing offers. As I'm only currently asserting that the appropriate API calls are being made.

Reply:
I started looking around for a different way and it turns out [how Django tests](https://github.com/django/django/blob/master/tests/) its runner would be useful to this plugin. More specifically how they [invoke the tests](https://github.com/django/django/blob/master/tests/runtests.py) and how the [test their Result classes](https://github.com/django/django/blob/master/tests/test_runner/test_debug_sql.py). 

So I tried out a simpler version of their testing method and it helps in bringing to the front of the code what exactly is being tested and test failures are clearer in the tracebacks/errors generated this way without having it going through Django's test runner unnecessarily.

Reply:
There are 2 separate things here:

1) Having a standard Django test suite which will exercise the example project and will provide all of the test results that this plugin can parse - pass, fail, expected pass or fail, sub-tests, etc.   This is very much like having the standard robotframework test suite inside the RF plugin. 

The result of ^^^ can be stand alone, e.g. printed on the terminal or we can make use of the current plugin to feed that result into Kiwi TCMS itself.   This is a high-level integration kind of thing because we don't really check if the plugin code itself is doing the right thing.  We expect the plugin to fail if something isn't right.

The majority of plugins are using similar kind of test suites where the plugin is only used to consume the results and feed them to the DB. If it doesn't crash then we assume it's fine.

2) You are looking into going into more details where you mock the backend API and assert that the corresponding methods have been called. In the other plugins we generally don't do this. There is some coverage for that in the tcms-api repository and the plugins are just consumers of the API on a higher level. 


I think we should not bother much with 2) at this moment. So far the existing method of implementing only 1) has worked well for us and we haven't had any complaints. When that changes we can reconsider adding another type of test suite like 2).

Reply:
Ahh, okay, I think I understand what we're going for here now. I'll get on with 1) I think I'll just close this PR then that implements something higher level.